net/procfs/net_tcp.c

```diff
@@ -0,0 +1,211 @@
+{
+  int addrlen = (domain == PF_INET) ?
+                INET_ADDRSTRLEN : INET6_ADDRSTRLEN;
+  FAR struct tcp_conn_s *conn = NULL;
+  char remote[addrlen + 1];
+  char local[addrlen + 1];
+  int len = 0;
+  void *laddr;
+  void *raddr;
+
+  net_lock();
+
+  if (priv->offset == 0)
+    {
+      len = snprintf(buffer, buflen, "TCP sl  "
+                                     "st flg ref tmr uack nrt   "
+#if CONFIG_NET_SEND_BUFSIZE > 0
+                                     "txsz   "
+#endif
+                                     "rxsz "
+                                     "%-*s "
+                                     "%-*s\n"
+                                     ,
+                                     INET6_ADDRSTRLEN / 2,
+                                     "local_address",
+                                     INET6_ADDRSTRLEN / 2,
+                                     "remote_address"
+                                     );
+      priv->offset = 1;
+    }
+
+  local[addrlen] = '\0';
+  remote[addrlen] = '\0';
+
+  while ((conn = tcp_nextconn(conn)) != NULL)
+    {
+#if defined(CONFIG_NET_IPv4) && defined(CONFIG_NET_IPv6)
+      if (conn->domain != domain)
+        {
+          continue;
+        }
+#endif /* CONFIG_NET_IPv4 && CONFIG_NET_IPv6 */
+
+      if (++(*skip) <= priv->offset)
+        {
+          continue;
+        }
+
+      if (buflen - len < TCP_LINELEN)
+        {
+          break;
+        }
+
+#if defined(CONFIG_NET_IPv4)
+      if (domain == PF_INET)
+        {
+          laddr = &conn->u.ipv4.laddr;
+          raddr = &conn->u.ipv4.raddr;
+        }
+#endif /* CONFIG_NET_IPv4 */
+
+#if defined(CONFIG_NET_IPv6)
+      if (domain == PF_INET6)
+        {
+          laddr = &conn->u.ipv6.laddr;
+          raddr = &conn->u.ipv6.raddr;
+        }
+#endif /* CONFIG_NET_IPv6 */
+
+      len += snprintf(buffer + len, buflen - len,
+                      "    %2" PRIu8
+                      ": %02" PRIx8
+                      " %3" PRIx8 " %3" PRIu8
+                      " %3" PRIu8
+#if defined(CONFIG_NET_TCP_WRITE_BUFFERS) || \
+    defined(CONFIG_NET_TCP_WINDOW_SCALE)
+                      " %4" PRIu32
+#else
+                      " %4" PRIu16
+#endif
+                      " %3" PRIu8
+#if CONFIG_NET_SEND_BUFSIZE > 0
+                      " %6" PRIu32
+#endif
+                      " %6u",
+                      priv->offset++,
+                      conn->tcpstateflags,
+                      conn->sconn.s_flags,
+                      conn->crefs,
+                      conn->timer,
+                      conn->tx_unacked,
+                      conn->nrtx,
+#if CONFIG_NET_SEND_BUFSIZE > 0
+                      tcp_wrbuffer_inqueue_size(conn),
+#endif
+                      (conn->readahead) ? conn->readahead->io_pktlen : 0);
+
+      len += snprintf(buffer + len, buflen - len,
+                      " %*s:%-6" PRIu16 " %*s:%-6" PRIu16 "\n",
+                      (domain == PF_INET6) ? addrlen / 2 : addrlen,
```

Comment:
why ipv6 need divide

Reply:
ipv6 will not occupy the whole line in some address, most of times only half the INET6_ADDRSTRLEN is enough, which will save the occupancy of formatting buffer